lib/connection/commands.js

```diff
@@ -419,9 +421,46 @@ var Response = function(bson, data, opts) {
   this.responseTo = data[this.index] | data[this.index + 1] << 8 | data[this.index + 2] << 16 | data[this.index + 3] << 24;
   this.index = this.index + 4;
 
-  // Skip op-code field
+  // Read op-code field
+  this.opCode = data[this.index] | data[this.index + 1] << 8 | data[this.index + 2] << 16 | data[this.index + 3] << 24;
   this.index = this.index + 4;
 
+  // If the message we get is OP_COMPRESSED, then we have to uncompress the data 
+  if (this.opCode == 2012) {
+    // Read original opcode
+    this.opCode = data[this.index] | data[this.index + 1] << 8 | data[this.index + 2] << 16 | data[this.index + 3] << 24;
+    this.index = this.index + 4;
+
+    // Read uncompressed size
+    var uncompressedSize = data[this.index] | data[this.index + 1] << 8 | data[this.index + 2] << 16 | data[this.index + 3] << 24;
+    this.index = this.index + 4;
+
+    // Read compressor ID
+    var compressorID = data[this.index];
+    this.index = this.index + 1;
+
+    // Read compressed data
+    var compressedData = data.slice(this.index)
+    switch (compressorID) {
+      case 0:
+        var uncompressedData = compressedData;
+        break;
+      case 1:
+        var uncompressedData = Snappy.uncompressSync(compressedData)
```

Comment:
Hmm, unfortunately I don't think this is going to cut it. We will need to use the async versions of `uncompress`/`unzip` instead, it's no good to block the main thread this way. We'll go over possible solutions to this tomorrow.